src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexFindOptimizations.cs

```diff
@@ -181,13 +181,13 @@ public RegexFindOptimizations(RegexTree tree, CultureInfo culture)
         public List<(char[]? Chars, string Set, int Distance, bool CaseInsensitive)>? FixedDistanceSets { get; }
 
         /// <summary>Try to advance to the next starting position that might be a location for a match.</summary>
-        /// <param name="text">The text to search.</param>
-        /// <param name="pos">The position in <paramref name="text"/>.  This is updated with the found position.</param>
-        /// <param name="beginning">The index in <paramref name="text"/> to consider the beginning for beginning anchor purposes.</param>
-        /// <param name="start">The index in <paramref name="text"/> to consider the start for start anchor purposes.</param>
-        /// <param name="end">The index in <paramref name="text"/> to consider the non-inclusive end of the string.</param>
+        /// <param name="textSpan">The text to search.</param>
+        /// <param name="pos">The position in <paramref name="textSpan"/>.  This is updated with the found position.</param>
+        /// <param name="beginning">The index in <paramref name="textSpan"/> to consider the beginning for beginning anchor purposes.</param>
+        /// <param name="start">The index in <paramref name="textSpan"/> to consider the start for start anchor purposes.</param>
+        /// <param name="end">The index in <paramref name="textSpan"/> to consider the non-inclusive end of the string.</param>
         /// <returns>true if a position to attempt a match was found; false if none was found.</returns>
-        public bool TryFindNextStartingPosition(string text, ref int pos, int beginning, int start, int end)
+        public bool TryFindNextStartingPosition(ReadOnlySpan<char> textSpan, ref int pos, int beginning, int start, int end)
```

Comment:
Can we change this method to not take start and end and instead ensure that the textSpan passed in has already been sliced accordingly?  That should simplify various aspects of the logic, and also likely help with some bounds check elimination, as we'll no longer be comparing indices to end but instead textSpan.Length in various places. The caller would also need to adjust the `ref int pos` in cases where start isn't 0.

We could also postpone doing this until a subsequent PR.

Reply:
I like that idea. However, is it ok if we push that to one of the later PRs? Reasoning being that this method is consumed by other engines which are not yet using spans, so my thinking is that once I move all the engines (or at least all of the ones consuming this API) to Spans, then this change will be easier to make. Does that work for you?

Reply:
> However, is it ok if we push that to one of the later PRs?

Yup!